daemon/create.go

```diff
@@ -21,25 +21,46 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+type createOpts struct {
+	params                  types.ContainerCreateConfig
+	managed                 bool
+	ignoreImagesArgsEscaped bool
+}
+
 // CreateManagedContainer creates a container that is managed by a Service
 func (daemon *Daemon) CreateManagedContainer(params types.ContainerCreateConfig) (containertypes.ContainerCreateCreatedBody, error) {
-	return daemon.containerCreate(params, true)
+	return daemon.containerCreate(createOpts{
+		params:                  params,
+		managed:                 true,
+		ignoreImagesArgsEscaped: false})
 }
 
 // ContainerCreate creates a regular container
 func (daemon *Daemon) ContainerCreate(params types.ContainerCreateConfig) (containertypes.ContainerCreateCreatedBody, error) {
-	return daemon.containerCreate(params, false)
+	return daemon.containerCreate(createOpts{
+		params:                  params,
+		managed:                 false,
+		ignoreImagesArgsEscaped: false})
+}
+
+// ContainerCreateIgnoreImagesArgsEscaped creates a regular container. This is called from the builder RUN case
+// and ensures that we do not take the images ArgsEscaped
+func (daemon *Daemon) ContainerCreateIgnoreImagesArgsEscaped(params types.ContainerCreateConfig) (containertypes.ContainerCreateCreatedBody, error) {
```

Comment:
Wary of adding another one of these things.

Reply:
Me too. But entirely unavoidable. I spent close on 3 weeks full-time working out how to do this so the end-to-end scenario works. I don't like it, but necessary. This was the only solution that I found to make it work outside of re-engineering docker to accept universally that Windows works through command lines, Linux works through args.

Reply:
Should we make the existing `ContainerCreate` accept a `ops ...` so that Windows can call it and pass an `WithIgnoreArgsEscaped()` option?

Reply:
Really, everything in here is terrible...  I'm fine adding this as is b/c such a change means several other unrelated changes.

I do have a branch where I'm working on rewriting all of this code and moving it all into a container service (like we have an image service and a volume service).